src/expressions/operators/types/CastAsOperator.ts

```diff
@@ -1,22 +1,23 @@
 import atomize from '../../dataTypes/atomize';
 import castToType from '../../dataTypes/castToType';
 import sequenceFactory from '../../dataTypes/sequenceFactory';
+import { ValueType } from '../../dataTypes/Value';
 import Expression from '../../Expression';
 
 class CastAsOperator extends Expression {
 	public _allowsEmptySequence: boolean;
 	public _expression: Expression;
-	public _targetType: string;
+	public _targetType: ValueType;
 
 	constructor(
 		expression: Expression,
 		targetType: { localName: string; namespaceURI: string | null; prefix: string },
 		allowsEmptySequence: boolean
 	) {
 		super(expression.specificity, [expression], { canBeStaticallyEvaluated: false });
-		this._targetType = targetType.prefix
+		this._targetType = (targetType.prefix
 			? `${targetType.prefix}:${targetType.localName}`
-			: targetType.localName;
+			: targetType.localName) as ValueType;
```

Comment:
Do we need to check whether this is in fact one of the allowed types?

Reply:
There are a few other instances of this in other files, grep for `as ValueType`

Reply:
it must be, the parser should pick up any errors